service/application.go

```diff
@@ -139,7 +141,11 @@ func New(params Parameters) (*Application, error) {
 	parserProvider := params.ParserProvider
 	if parserProvider == nil {
 		// use default provider.
-		parserProvider = parserprovider.Default()
+		logger, err := zap.NewDevelopment() // TODO find a way to share with app.
+		if err != nil {
+			return nil, err
+		}
+		parserProvider = parserprovider.NewConfigSourceParserProvider(logger, params.ApplicationStartInfo)
```

Comment:
Since this was just my manual tests the one using the config sources is the default but this is not what is wanted in practice.